app/main.py

```diff
@@ -1,213 +1,49 @@
-    for a in arthur["armour"]:
-        arthur["protection"] += a["protection"]
-
-    # apply weapon
-    arthur["power"] += arthur["weapon"]["power"]
-
-    # apply potion if exist
-    if arthur["potion"] is not None:
-        if "power" in arthur["potion"]["effect"]:
-            arthur["power"] += arthur["potion"]["effect"]["power"]
-
-        if "protection" in arthur["potion"]["effect"]:
-            arthur["protection"] += arthur["potion"]["effect"]["protection"]
-
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
-
-    # mordred
-    mordred = knightsConfig["mordred"]
-
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
-
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
-
-    # apply potion if exist
-    if mordred["potion"] is not None:
-        if "power" in mordred["potion"]["effect"]:
-            mordred["power"] += mordred["potion"]["effect"]["power"]
-
-        if "protection" in mordred["potion"]["effect"]:
-            mordred["protection"] += mordred["potion"]["effect"]["protection"]
-
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
-
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
-
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
-
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
+    list_of_knights = {}
+    for knight in knights_config:
+        list_of_knights[knight] = Knight(
+            knights_config[knight]["name"],
+            knights_config[knight]["power"],
+            knights_config[knight]["hp"],
+            knights_config[knight]["armour"],
+            knights_config[knight]["weapon"],
+            knights_config[knight]["potion"]
+        )
+
+    # Create lancelot and preparation to fight
+    lancelot = list_of_knights["lancelot"]
+    preparation_to_fight(lancelot)
+
+    # Create arthur and preparation to fight
+    arthur = list_of_knights["arthur"]
+    preparation_to_fight(arthur)
+
+    # Create mordred and preparation to fight
+    mordred = list_of_knights["mordred"]
+    preparation_to_fight(mordred)
+
+    # Create red_knight and preparation to fight
+    red_knight = list_of_knights["red_knight"]
+    preparation_to_fight(red_knight)
```

Comment:
The preparation phase for each knight is repetitive. Consider using a loop to iterate over the `list_of_knights` dictionary and call `preparation_to_fight` for each knight. This will make the code more concise and easier to maintain.